lib/linux_admin/ssh.rb

```diff
@@ -0,0 +1,102 @@
+module LinuxAdmin
+  class SSH
+    attr_reader :ip
+    attr_reader :username
+    attr_reader :private_key
+    attr_reader :agent
+
+    def initialize(ip, username, private_key)
+      @ip = ip
+      @private_key = private_key
+      @username = username
+    end
+
+    def perform_commands(commands = [], options = {})
+      result = nil
+      socket = nil
+      if options[:agent_enabled]
+        @agent = LinuxAdmin::SSHAgent.new(@private_key, :agent_socket => options[:agent_socket])
+        @agent.with_service do |socket|
+          result = if block_given?
+                     execute_commands(commands, :socket => socket, :stdin => options[:stdin], &Proc.new)
```

Comment:
Are you trying to pass the given block to `execute_commands` here?

If so, I think you would need to add `&block` to the params list of `perform_commands` and use `&block` where you have `&Proc.new` now.